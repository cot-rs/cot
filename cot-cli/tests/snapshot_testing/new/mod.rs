use clap_verbosity_flag::{OffLevel, Verbosity};

use super::*;

#[test]
#[expect(clippy::cast_possible_truncation)]
fn create_new_project() {
    let cmd = cot_cli!("new");
    for (idx, ref mut cli) in cot_clis_with_verbosity(&cmd).into_iter().enumerate() {
        let tempdir = tempfile::TempDir::with_prefix("cot-test-").unwrap();
        let filter = Verbosity::<OffLevel>::new(idx as u8, 0).filter();

        insta::with_settings!(
            {
                description => format!("Verbosity level: {filter}"),
                filters => [GENERIC_FILTERS, TEMP_PATH_FILTERS, TEMP_PROJECT_FILTERS].concat()
            },
            {
            assert_cmd_snapshot!(cli.arg(tempdir.path().join("project")));
            }
        );
    }
}

#[test]
#[expect(clippy::cast_possible_truncation)]
fn create_new_project_with_custom_name() {
    let cmd = cot_cli!("new", "--name", "my_project");
    for (idx, ref mut cli) in cot_clis_with_verbosity(&cmd).into_iter().enumerate() {
        let tempdir = tempfile::TempDir::with_prefix("cot-test-").unwrap();
        let filter = Verbosity::<OffLevel>::new(idx as u8, 0).filter();

        insta::with_settings!(
            {
                description => format!("Verbosity level: {filter}"),
                filters => [GENERIC_FILTERS, TEMP_PATH_FILTERS, TEMP_PROJECT_FILTERS].concat()
            },
            {
            assert_cmd_snapshot!(cli.arg(tempdir.path().join("project")));
            }
        );
    }
}
