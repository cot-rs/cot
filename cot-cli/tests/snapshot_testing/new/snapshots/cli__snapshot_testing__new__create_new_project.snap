---
source: cot-cli/tests/snapshot_testing/new/mod.rs
description: "Verbosity level: off"
info:
  program: cot
  args:
    - new
    - "-q"
    - /tmp/cot-test-tcwDL5/project
---
success: true
exit_code: 0
----- stdout -----

----- stderr -----
[1m[92m    Creating[0m Cot project `project`
[1m[32m     Created[0m Cot project `project`
